packages/rocketchat-api/server/v1/channels.js

```diff
@@ -553,38 +547,32 @@ RocketChat.API.v1.addRoute('channels.members', { authRequired: true }, {
 	get() {
 		const findResult = findChannelByIdOrName({
 			params: this.requestParams(),
-			checkedArchived: false,
-			returnUsernames: true
+			checkedArchived: false
 		});
 
 		if (findResult.broadcast && !RocketChat.authz.hasPermission(this.userId, 'view-broadcast-member-list')) {
 			return RocketChat.API.v1.unauthorized();
 		}
 
 		const { offset, count } = this.getPaginationItems();
-		const { sort } = this.parseJsonQuery();
+		const { sort = {} } = this.parseJsonQuery();
 
-		const shouldBeOrderedDesc = Match.test(sort, Object) && Match.test(sort.username, Number) && sort.username === -1;
-
-		let members = RocketChat.models.Rooms.processQueryOptionsOnResult(Array.from(findResult.usernames).sort(), {
+		const members = RocketChat.models.Subscriptions.findByRoomId(findResult._id, {
+			sort: {'u.username':  sort.username != null ? sort.username : 1},
 			skip: offset,
 			limit: count
-		});
-
-		if (shouldBeOrderedDesc) {
-			members = members.reverse();
-		}
+		}).fetch().map(s => s.u && s.u.username);
```

Comment:
the next query will search for all users without usernames

Reply:
actually the issue was it returning subscriptions with `null` usernames. I've fixed it in 187dbc527aa078a848e60998c2cb6cce3cde5f82